targets/alert-cli/src/diff/dila/Agreement/__tests__/processAgreementFileChanges.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { describe, expect } from "@jest/globals";
+import type { ConvenientPatch, Tree } from "nodegit";
+
+import processAgreementFileChanges from "../ProcessAgreementFileChanges";
+
+describe("Creation des AgreementFileChange à partir d'un patch", () => {
+  const createMockPatch = (
+    added: boolean,
+    removed: boolean,
+    modified: boolean
+  ) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    const newFile: DiffFile = { path: () => "my-file.json" };
+    return {
+      isAdded: () => added,
+      isDeleted: () => removed,
+      isModified: () => modified,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      newFile: () => newFile,
+    };
+  };
```

Comment:
Pour simplifier l'écriture du test ensuite et ne pas avoir les `disable` sur les règles, tu peux forcer le type directement ici :
```suggestion
const createPatch = (
    added: boolean,
    removed: boolean,
    modified: boolean
  ): ConvenientPatch => {
    const newFile = { path: () => "my-file.json" } as DiffFile;
    return {
      isAdded: () => added,
      isDeleted: () => removed,
      isModified: () => modified,
      newFile: () => newFile,
    } as ConvenientPatch;
  };
```

D'ailleurs la fonction ne crée pas un mock mais simplement un patch (une donnée pour le jeu de test) d'où le renommage de la fonction :)